exporter/metric/metric.go

```diff
@@ -186,7 +187,7 @@ func (me *metricExporter) ExportMetrics(ctx context.Context, cps export.Checkpoi
 // if the descriptor is not registered in Cloud Monitoring yet.
 func (me *metricExporter) exportMetricDescriptor(ctx context.Context, cps export.CheckpointSet) error {
 	mds := make(map[key]*googlemetricpb.MetricDescriptor)
-	aggError := cps.ForEach(func(r export.Record) error {
+	aggError := cps.ForEach(export.CumulativeExporter, func(r export.Record) error {
```

Comment:
Technically we always export `Cumulatives`, but this is not how the interfaces are intended to be used. The interface design here is a little odd / confusing and I wouldn't be surprised if this changes before GA.

The first argument to `ForEach` is an `ExportSelector`. You should pass through `me` itself here as the exporter already implements this interface, rather than passing in this constant selector (likewise for each of the calls to `ForEach` below). This will mean the same function commented above will be used, rather than us having to specify `Cumulative` in multiple places here.

Reply:
Hm, it doesn't seem to think `metricExporter` implements the `ExportKindSelector` interface:
```
../../exporter/metric/metric.go:190:26: cannot use *me (type metricExporter) as type "go.opentelemetry.io/otel/sdk/export/metric".ExportKindSelector in argument to cps.ForEach:
	metricExporter does not implement "go.opentelemetry.io/otel/sdk/export/metric".ExportKindSelector (missing ExportKindFor method)

```

Reply:
Hmm yea this is a bit weird because we have the Export wrapper. You implemented the `ExportKindSelector` above (https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/73/files#diff-61301ca921134b70b15b20369de304f1R90) but its for the `Exporter` type rather than the `metricExporter` type

This might warrant some more significant refactoring but for now you could just pass the wrapper exporter into the `newMetricExporter` constructor so that you can use it here.